framework/src/play/templates/TemplateLoader.java

```diff
@@ -74,25 +79,39 @@ public static Template load(String path) {
 
     URL fromClasspath = Thread.currentThread().getContextClassLoader().getResource("views/" + path);
     if (fromClasspath != null) {
-      return loadTemplateFromClasspath(path, fromClasspath);
+      return getTemplateFromClasspath(path, fromClasspath);
     }
 
     URL resource = Thread.currentThread().getContextClassLoader().getResource(path);
     if (resource != null) {
-      return loadTemplateFromClasspath(path, resource);
+      return getTemplateFromClasspath(path, resource);
     }
 
     throw new TemplateNotFoundException(path);
   }
 
-  private static Template loadTemplateFromClasspath(String path, URL resource) {
+  private static Template getTemplateFromClasspath(String path, URL resource) {
     File templateFile = new File(Play.tmpDir, path);
+    if (!canBeReused(templateFile)) {
+      loadTemplateFromClasspath(templateFile, resource);
+    }
+    return TemplateLoader.load(templateFile);
+  }
+
+  private static boolean canBeReused(File templateFile) {
+    return templateFile.exists() &&
+        currentTimeMillis() - templateFile.lastModified() < TEMPLATE_CACHE_TIMEOUT_MS;
+  }
+
+  private static synchronized void loadTemplateFromClasspath(File templateFile, URL resource) {
+    if (canBeReused(templateFile)) return;
+
+    logger.info("Loading template to {} from {}", templateFile, resource);
     try {
-      copyURLToFile(resource, templateFile);
+      copyURLToFile(resource, templateFile, TEMPLATE_CONNECT_TIMEOUT_MS, TEMPLATE_READ_TIMEOUT_MS);
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
-    return TemplateLoader.load(Play.file("tmp/" + path));
```

Comment:
For the record: this commit (unintentionally? :slightly_smiling_face: ) fixed and old bug introduced by commit cfe9ff10df568cfdf42a4e3b9f81dbf89124956c:
> When `Play.tmpDir` differs from the default "tmp", then loading templates
from the classpath fails miserably.